File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();

        int arr[] = new int[101];
        int tmp[] = new int[101];

        for (int i = 1; i <= n; i++) arr[i] = i;

        for (int i = 0; i < m; i++) {
            int start = sc.nextInt();
            int end = sc.nextInt();

            for (int k = start; k <= end; k++) tmp[end-(k-start)] = arr[k];
            for (int k = start; k <= end; k++) arr[k] = tmp[k];
        }

        for (int i = 1; i <= n; i++) System.out.print(arr[i] + " ");
    }
}

File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.cc
#include <stdio.h>

int main(void) {
    int n, m;
    scanf("%d %d", &n, &m);
    
    int arr[101] = {0,};
    int tmp[101] = {0,};
    for (int i = 1; i <= n; i++) arr[i] = i;

    while(m--) {
        int start, end;
        scanf("%d %d", &start, &end);

        for (int i = start; i <= end; i++) tmp[end-(i-start)] = arr[i];
        for (int i = start; i <= end; i++) arr[i] = tmp[i];
    }

    for (int i = 1; i <= n; i++) printf("%d ", arr[i]);
    return 0;
}